### PR TITLE
Refactor front matter migration

### PR DESCRIPTION
**Pull request by hamelsmu**
cc: @jph00 @seeM this PR is not complete, but I'm opening it early to get some feedback on the direction on the refactor.

The next thing would be to refactor `NBProcessor` but I can do that in a follow on PR

### CODE REVIEW

File: nbs/01_read.ipynb [hunk 1]
```diff
@@ -35,10 +35,10 @@
     "from fastcore.xdg import *\n",
```
Comment (seeM) [inline nbs/01_read.ipynb:35]: <pre class="ql-syntax" spellcheck="false">Line #49.                              {'title':yaml_str(title)}, {'description':yaml_str(desc)} if desc else {})
</pre><p>Does <code>yaml_str</code> solve a specific problem here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: nbs/09_processors.ipynb [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "#|export\n",
```
Comment (seeM) [inline nbs/09_processors.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #3.    print(_res.text[0])
</pre><p>Might be more expressive if NB getitem indexed into cells? Then you could do <code>print(_res[0].source)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: nbs/15_migrate.ipynb [hunk 3]
```diff
@@ -28,7 +28,8 @@
    "source": [
```
Comment (seeM) [inline nbs/15_migrate.ipynb:28]: <pre class="ql-syntax" spellcheck="false">Line #2.    def _cat_slug(d):
</pre><p>In the spirit of being concise, I'd personally prefer <code>d</code> as the param name over <code>fmdict</code> in the other notebooks too </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (seeM) [inline nbs/15_migrate.ipynb:28]: <pre class="ql-syntax" spellcheck="false">Line #4.        slug = '/'.join(sorted(d.get('categories', '')))
</pre><p>Should this default to an empty list?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (seeM) [inline nbs/15_migrate.ipynb:28]: <pre class="ql-syntax" spellcheck="false">Line #4.    def _alias(fm:dict, p:Path):
</pre><p>I'd also prefer <code>fm</code> instead of <code>fmdict</code>  in the other notebooks too - more concise</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (seeM) [inline nbs/15_migrate.ipynb:28]: <pre class="ql-syntax" spellcheck="false">Line #7.                ss = listify(cell['source'])
</pre><p>Why are these needed? Shouldn't it be handled by NbCell?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/nbdev/pull/803/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

File: nbs/09_processors.ipynb (hunk 2 above)
Comment (hamelsmu) [reply nbs/09_processors.ipynb:29]: Love this idea

File: nbs/01_read.ipynb (hunk 1 above)
Comment (hamelsmu) [reply nbs/01_read.ipynb:35]: Yes, this is dealing with quotes, since we are extracting the YAML with regex we need to also make sure its valid YAML

File: nbs/15_migrate.ipynb (hunk 3 above)
Comment (hamelsmu) [reply nbs/15_migrate.ipynb:28]: Thanks yes this a bit obtuse, not necessary, thanks for flagging this.

File: nbs/09_processors.ipynb (hunk 2 above)
Comment (hamelsmu) [reply nbs/09_processors.ipynb:29]: Unfortunately `__getitem__` seems to interfere with AttrDict  (posted a screenshot in Skype)
